crates/services/executor/src/executor.rs

```diff
@@ -1030,228 +1080,671 @@ where
+        <Tx as IntoChecked>::Metadata: CheckedMetadata,
+        T: KeyValueInspect<Column = Column>,
+    {
+        let tx_id = checked_tx.id();
+
+        if self.options.utxo_validation {
+            checked_tx = self.validate_utxos(checked_tx, header, tx_st_transaction)?;
+        }
+
+        self.validate_inputs_state(
+            checked_tx.transaction().inputs(),
+            tx_id,
+            tx_st_transaction,
+        )?;
+
+        let (reverted, state, mut tx, receipts) = self.attempt_tx_execution_with_vm(
+            &mut checked_tx,
+            &header,
+            coinbase_contract_id,
+            gas_price,
+            tx_st_transaction,
+        )?;
+
+        let (used_gas, tx_fee) = self.total_fee_paid(&tx, &receipts, gas_price)?;
+
+        self.spend_input_utxos(tx.inputs(), tx_st_transaction, reverted, execution_data)?;
+
+        self.persist_output_utxos(
+            *header.height(),
+            execution_data,
+            &tx_id,
+            tx_st_transaction,
+            tx.inputs(),
+            tx.outputs(),
+        )?;
+
+        self.update_tx_outputs(tx_st_transaction, tx_id, &mut tx)?;
+
+        Self::validate_tx_outcome(&tx, &checked_tx)?;
+
+        tx_st_transaction
+            .storage::<ProcessedTransactions>()
+            .insert(&tx_id, &())?;
+
+        Self::update_execution_data(
+            execution_data,
+            receipts,
+            used_gas,
+            tx_fee,
+            reverted,
+            state,
+            tx_id,
+        )?;
+
+        Ok(tx.into())
+    }
+
+    fn validate_tx_outcome<Tx: PartialEq + IntoChecked + UniqueIdentifier>(
+        tx: &Tx,
+        checked_tx: &Checked<Tx>,
+    ) -> ExecutorResult<()> {
+        if tx != checked_tx.transaction() {
+            return Err(ExecutorError::InvalidTransactionOutcome {
+                transaction_id: checked_tx.id(),
+            })
         }
+        Ok(())
+    }
 
-        let tx = mint.into();
+    fn update_tx_outputs<Tx, T>(
+        &self,
+        tx_st_transaction: &mut StorageTransaction<T>,
+        tx_id: TxId,
+        tx: &mut Tx,
+    ) -> ExecutorResult<()>
+    where
+        Tx: ExecutableTransaction + PartialEq + Cacheable + Send + Sync + 'static,
+        <Tx as IntoChecked>::Metadata: CheckedMetadata,
+        T: KeyValueInspect<Column = Column>,
+    {
+        let mut outputs = core::mem::take(tx.outputs_mut());
+        self.compute_state_of_not_utxo_outputs(
+            &mut outputs,
+            tx.inputs(),
+            tx_id,
+            tx_st_transaction,
+        )?;
+        *tx.outputs_mut() = outputs;
+        Ok(())
+    }
+
+    fn update_execution_data(
+        execution_data: &mut ExecutionData,
+        receipts: Vec<Receipt>,
+        used_gas: Word,
+        tx_fee: Word,
+        reverted: bool,
+        state: ProgramState,
+        tx_id: TxId,
+    ) -> ExecutorResult<()> {
+        execution_data.coinbase = execution_data
```

Comment:
It seems you can call `let (used_gas, tx_fee) = self.total_fee_paid(&tx, &receipts, gas_price)?;` inside of this function instead of passing arguments